Add best practices documentation and update other docs (#3938)
Azure/azure-service-operator: docs/hugo/content/guide/asov1-asov2-migration/_index.md

```
should show `v1` resources.

{{% alert title="Warning" color="warning" %}}

> Make sure that each ASOv1 resource you delete has the `skipreconcile=true` annotation before you delete it

Reviewer: ```suggestion
Ensure that you've verified each ASOv1 resource you delete has the `skipreconcile=true` annotation before you delete it
```

Once you have annotated the resources, double-check that they all have the `skipreconcile` annotation. 

{{% alert title="Warning" color="warning" %}}

> If you delete an ASOv1 resource that does not have this annotation it will delete the underlying

Reviewer: ```suggestion
If you delete an ASOv1 resource that does not have this annotation it _**will**_ delete the underlying
```

Once you have `resources.yaml` locally, examine it to ensure that it has the resources you expect.

{{% alert title="Note" %}}

> `asoctl` is importing the resources from Azure. This means it will not preserve any Kubernetes labels/annotations

Reviewer: It's not a policy choice, but a limitation, because those annotations aren't present in Azure. Maybe we need to be more explicit about this?

```suggestion
`asoctl` is importing the resources from Azure. This means it cannot preserve any Kubernetes labels/annotations
```

rules. ASOv1 always exports these secrets, there is no user configuration required on either the name of the secret or its values.
In contrast, ASOv2 requires the user to opt-in to secret export, via the `spec.operatorSpec.secrets` property.

{{% alert title="Note" %}}

> ASOv2 also splits secrets/configuration into two categories: Secrets provided by you, and secrets generated

Reviewer: You mention configuration up front, but then don't mention it at all.

Either document config as well, or remove it altogether.

Author: Dropped it as ASOv1 doesn't do ConfigMaps and so from a migration perspective only secrets is interesting here. ConfigMap export is a new feature that they can use in ASOv2 but the samples and other docs can help them discover that, focus here should be on migration.

`-annotation serviceoperator.azure.com/reconcile-policy=skip` argument or manually by modifying `resources.yaml` 
afterward.

{{% alert title="Warning" color="warning" %}}
Be _very_ careful issuing deletions of ASOv2 resources once you've imported them if they do not have

> the serviceoperator.azure.com/reconcile-policy=skip annotation. Just like ASOv1,
> ASOv2 will delete the underlying Azure resource by default!

Reviewer: ```suggestion
the `serviceoperator.azure.com/reconcile-policy=skip` annotation. Just like ASOv1,
ASOv2 _**will**_ delete the underlying Azure resource by default!
```
